sem_covid/entrypoints/etl_dags/finreg_worker.py

```diff
@@ -0,0 +1,385 @@
+        for html_manifestation in json_content.get('htmls_to_download'):
+            filename = hashlib.sha256(html_manifestation.encode('utf-8')).hexdigest()
+
+            logger.info(f"Downloading HTML manifestation for {(json_content['title'] or json_content['work'])}")
+
+            html_file = filename + '_html.zip'
+            if download_file(json_content, html_manifestation, html_file, minio):
+                counter['html'] += 1
+    elif json_content.get('manifs_pdf'):
+        for pdf_manifestation in json_content.get('pdfs_to_download'):
+
+            filename = hashlib.sha256(pdf_manifestation.encode('utf-8')).hexdigest()
+
+            logger.info(f"Downloading PDF manifestation for {(json_content['title'] or json_content['work'])}")
+
+            pdf_file = filename + '_pdf.zip'
+            if download_file(json_content, pdf_manifestation, pdf_file, minio):
+                counter['pdf'] += 1
+    else:
+        logger.exception(f"No manifestation has been found for {json_content['title']}")
+
+    minio.put_object(json_filename, json.dumps(json_content))
+
+    logger.info(f"Downloaded {counter['html']} HTML manifestations and {counter['pdf']} PDF manifestations.")
+
+
+def upload_to_elastic_callable(**context):
+    if "work" not in context['dag_run'].conf:
+        logger.error(
+            "Could not find the work in the provided configuration. This DAG is to be triggered by its parent only.")
+        return
+
+    work = context['dag_run'].conf['work']
+    json_file_name = FIELD_DATA_PREFIX + hashlib.sha256(work.encode('utf-8')).hexdigest() + ".json"
+    es_adapter = StoreRegistry.es_index_store()
+    logger.info(f'Using ElasticSearch at {config.ELASTICSEARCH_HOST_NAME}:{config.ELASTICSEARCH_PORT}')
+
+    logger.info(f'Loading files from {config.MINIO_URL}')
+
+    minio = StoreRegistry.minio_object_store(config.EU_FINREG_CELLAR_BUCKET_NAME)
+    json_content = json.loads(minio.get_object(json_file_name).decode('utf-8'))
+    try:
+        logger.info(
+            f'Sending to ElasticSearch ( {config.EU_FINREG_CELLAR_ELASTIC_SEARCH_INDEX_NAME} ) the file {json_file_name}')
+        es_adapter.index(index_name=config.EU_FINREG_CELLAR_ELASTIC_SEARCH_INDEX_NAME,
+                         document_id=hashlib.sha256(work.encode('utf-8')).hexdigest(),
+                         document_body=json_content)
+    except Exception as ex:
+        logger.exception(ex)
+
+    logger.info(f'Sent {json_file_name} file(s) to ElasticSearch.')
+
+
+default_args = {
+    "owner": "airflow",
+    "depends_on_past": False,
+    "start_date": datetime(2021, 2, 16),
+    "email": ["mclaurentiu79@gmail.com"],
+    "email_on_failure": False,
+    "email_on_retry": False,
+    "retries": 1,
+    "retry_delay": timedelta(minutes=500)
+}
+
+with DAG(DAG_NAME, default_args=default_args, schedule_interval=None, max_active_runs=4, concurrency=4) as dag:
+    download_documents_and_enrich_json = PythonOperator(
```

Comment:
let's retry a few times, with a timeout in between them. 
Querying the cellar may be failing sometimes.

Reply:
what should be retried here? the dag ?

Reply:
what do you think?